doc/tutorial/tutorial_lfpykit.rst

```diff
@@ -0,0 +1,159 @@
+.. _tutorial_lfpykit:
+
+Extracellular signals
+=====================
+
+This example takes different elements from the above example tutorials
+(:ref:`A simple single cell model <tutorialsinglecell>`,
+:ref:`A detailed single cell recipe <tutorialsinglecellswcrecipe>`)
+to create a geometrically
+complex single cell model from an SWC morphology file, and adds predictions of
+extracellular potentials using the external `LFPykit <https://lfpykit.readthedocs.io/en/latest>`_ Python library
+(https://LFPykit.readthedocs.io, https://github.com/LFPy/LFPykit).
+`LFPykit <https://lfpykit.readthedocs.io/en/latest>`_ provides a few different classes facilitating
+calculations of extracellular potentials and related electroencephalography (EEG)
+and magnetoencephalography (MEG) signals from geometrically detailed neuron models under various assumptions.
+These are signals that mainly stem from transmembrane currents.
+
+This tutorial goes through the required steps in more detail.
+
+In this tutorial, the neuron model itself is kept deliberately simple with only
+passive (leaky) membrane dynamics, and it receives sinusoid synaptic current
+input in one arbitrary chosen control volume (CV).
+
+.. Note::
+
+   **Concepts covered in this tutorial:**
+
+   1. Building a morphology from an SWC file.
+   2. Recording of transmembrane currents using :class:`arbor.cable_probe_total_current_cell`
+   3. Recording of stimulus currents using :class:`arbor.cable_probe_stimulus_current_cell`
+   4. Calling the :class:`arbor.place_pwlin` API
+   5. Map recorded transmembrane currents to extracellular potentials using `LFPykit <https://lfpykit.readthedocs.io/en/latest>`_
+
+
+.. _tutorial_lfpykit-model:
+
+The single cell model
+---------------------
+
+First we import some required modules:
+
+.. literalinclude:: ../../python/example/single_cell_extracellular_potentials.py
+   :language: python
+   :lines: 16-18
+
+
+Define ``Recipe`` class:
+
+.. literalinclude:: ../../python/example/single_cell_extracellular_potentials.py
+   :language: python
+   :lines: 24-56
+
+
+Load morphology on ``SWC`` file format:
+
+.. code-block:: python
+
+   # define morphology needed for ``arbor.place_pwlin`` and ``arbor.cable_cell`` below
```

Comment:
In other examples, the cell related stuff is generated from inside a function, providing some clarify in the code. I would argue to do that here too. It's lines 58-107; the function could return `p,cell`.

Also, is there anything to describe the cell textually? For instance, you add the `iclamp` in a specific place, I think this is quite tied to `single_cell_detailed.swc`. If so, then hardcode the filename + consider adding an illustration of that location (see `doc/concepts/labels.rst`, I can do it if needed.)

Reply:
I've added a callable function `make_cable_cell` returning `p, cell` as suggested. 
If you see a neater way of setting a stimulus location as `clamp_location = arbor.location(4, 1/6)`, please revise. 
The location is now added to the results figure.